core/src/main/scala/org/apache/spark/deploy/master/MasterArguments.scala

```diff
@@ -20,18 +20,24 @@ package org.apache.spark.deploy.master
 import scala.annotation.tailrec
 
 import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
 import org.apache.spark.util.{IntParam, Utils}
 
 /**
  * Command-line parser for the master.
  */
-private[master] class MasterArguments(args: Array[String], conf: SparkConf) {
+private[master] class MasterArguments(args: Array[String], conf: SparkConf) extends Logging {
   var host = Utils.localHostName()
   var port = 7077
   var webUiPort = 8080
   var propertiesFile: String = null
 
   // Check for settings in environment variables
+  if (System.getenv("SPARK_MASTER_IP") != null) {
+    logWarning("SPARK_MASTER_IP is deprecated, please use SPARK_MASTER_HOST")
+    host = System.getenv("SPARK_MASTER_IP")
+  }
+
   if (System.getenv("SPARK_MASTER_HOST") != null) {
```

Comment:
This looks good to me. Now, one final thought. We are sort of moving away from env variables anyway. I think we could now remove the handling of `SPARK_MASTER_HOST` here since it's redundant with the scripts in `sbin/`.

And then beyond that, I wonder if we should just handle the deprecation warning in the same place, in the scripts, rather than code. What do you think?